main.tf

```diff
@@ -1,99 +1,224 @@
+  cloudwatch_log_group_name = "fargate/${var.app_name}"                 // CloudWatch Log Group name
+
+  container_definitions = [
+    for def in var.container_definitions : {
+      name       = def.name
+      image      = def.image
+      essential  = true
+      privileged = false
+      portMappings = [
+        for port in def.ports :
+        {
+          containerPort = port
+          hostPort      = port
+          protocol      = "tcp"
+        }
+      ]
+      logConfiguration = {
+        logDriver = "awslogs"
+        options = {
+          awslogs-group         = local.cloudwatch_log_group_name
+          awslogs-region        = data.aws_region.current.name
+          awslogs-stream-prefix = def.name
+        }
       }
+      environment = [
+        for key in keys(def.environment_variables) :
+        {
+          name  = key
+          value = lookup(def.environment_variables, key)
+        }
+      ]
+      secrets = [
+        for key in keys(def.secrets) :
+        {
+          name      = key
+          valueFrom = "${local.ssm_parameter_arn_base}${replace(lookup(def.secrets, key), "/^//", "")}"
+        }
+      ]
+      mountPoints = []
+      volumesFrom = []
     }
+  ]
+}
+
+# ==================== ALB ====================
+resource "aws_alb" "alb" {
+  name            = local.alb_name
+  subnets         = var.public_subnet_ids
+  security_groups = [aws_security_group.alb-sg.id]
+  tags            = var.tags
+}
+resource "aws_security_group" "alb-sg" {
+  name        = "${local.alb_name}-sg"
+  description = "Controls access to the ${local.alb_name}"
+  vpc_id      = var.vpc_id
+
+  // allow access to the ALB from anywhere for 80 and 443
+  ingress {
+    from_port   = 80
+    to_port     = 80
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
   }
-  listeners = {
-    80 = {
-      protocol              = "HTTP"
-      https_certificate_arn = null
-      redirect_to = {
-        host     = null
-        path     = null
-        port     = 443
-        protocol = "HTTPS"
-      }
-      forward_to = null
-    },
-    443 = {
-      protocol              = "HTTPS"
-      https_certificate_arn = module.acs.certificate.arn
-      redirect_to           = null
-      forward_to = {
-        target_group   = "blue"
-        ignore_changes = true
-      }
+  ingress {
+    from_port   = 443
+    to_port     = 443
+    protocol    = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+  // allow any outgoing traffic
+  egress {
+    protocol    = "-1"
+    from_port   = 0
+    to_port     = 0
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+  tags = var.tags
+}
+resource "aws_alb_target_group" "blue" {
+  name     = "${var.app_name}-tgb"
+  port     = var.image_port
+  protocol = "HTTP"
+  vpc_id   = var.vpc_id
+
+  target_type          = "ip"
+  deregistration_delay = var.target_group_deregistration_delay
+  health_check {
+    path                = var.health_check_path
+    interval            = 30
```

Comment:
I think at least some of these need to be passed through variables, even if they are defaulted.

Reply:
Done